src/app/3d/qgs3dmapcanvaswidget.cpp

```diff
@@ -1168,6 +1177,58 @@ void Qgs3DMapCanvasWidget::setSceneExtent( const QgsRectangle &extent )
     mMainCanvas->unsetMapTool( mMapToolExtent.get() );
 }
 
+void Qgs3DMapCanvasWidget::setClippingPlanesOn2DCanvas()
+{
+  if ( !qobject_cast<QgsMapToolClippingPlanes *>( mMainCanvas->mapTool() ) )
+    mMapToolPrevious = mMainCanvas->mapTool();
+
+  mMainCanvas->setMapTool( mMapToolClippingPlanes.get() );
+  QgisApp::instance()->activateWindow();
+  QgisApp::instance()->raise();
+  mMessageBar->pushInfo( QString(), tr( "Select a rectangle using 3 points on the main 2D map view to define the cross-section of this 3D scene" ) );
+}
+
+void Qgs3DMapCanvasWidget::setViewOnClippingPlanesChanged( const QgsVector3D &startPoint, const QgsVector3D &endPoint, const double width )
```

Comment:
It looks like you changed from a `slot` to a function but you haven't changed its name. The current name does not make a lot of sense to me. 

I would prefer something like `enableClippingPlanes`. In that case, it would take 2 arguments: the list of clipping planes and the new camera. Which means that the clipping planes and the new QgsCameraPose should be computed in `QgsMapToolClippingPlanes`.